### PR TITLE
Create new user data service to seperate data calls from auth service 

### PR DESCRIPTION
**Pull request by srknzl**
- A new user data service that inherits data service is created.
- All the usages of auth service related user data are transferred to user data service newly created.

### CODE REVIEW

File: frontend/src/app/data/providers/data/data.service.ts [hunk 1]
```diff
@@ -14,9 +14,8 @@ export abstract class DataService<T> {
     return '/api/' + this.api + '/' + id;
   }
 
-  get(id: string = '', params: { limit?: number, skip?: number, search?: string } = {}): Observable<T | T[]> {
+  get(id: string = '', params: { limit?: number, skip?: number, search?: string, isUser?: string} = {}): Observable<T | T[]> {
```
Comment (serkan-ozel-2015400123) [inline frontend/src/app/data/providers/data/data.service.ts]: right now in feature/backend/search branch @nazmicancalik uses search by taking "isUser" boolean query parameter. Based on that I add this.

File: frontend/src/app/events/components/search/search.component.ts [hunk 2]
```diff
@@ -55,9 +63,29 @@ export class SearchComponent implements OnInit {
     alert('Searching between times: ' + this.fromTime.value+ ' and ' + this.toTime.value);
   }
   search(){
+    this.loading = true;
```
Comment (serkan-ozel-2015400123) [inline frontend/src/app/events/components/search/search.component.ts]: For loading spinner small changes are added,  to finalize loading by covering all pages in the future.

File: frontend/src/app/events/components/search/search.component.ts [hunk 3]
```diff
@@ -30,6 +33,10 @@ export class SearchComponent implements OnInit {
     ,'Turkish Folk Music'
     ,'Concert'
   ];
+
+  foundUsers: User[] | User = [];
```
Comment (serkan-ozel-2015400123) [inline frontend/src/app/events/components/search/search.component.ts]: Return type of get calls in data services are in the format T | T[]

File: frontend/src/app/events/pages/event/event.page.ts [hunk 4]
```diff
@@ -4,10 +4,9 @@ import { ActivatedRoute } from '@angular/router';
 import { EventService } from '../../../data/providers/event/event.service';
 
 import {AlertController, LoadingController} from "@ionic/angular";
-import {HttpErrorResponse} from "@angular/common/http";
```
Comment (serkan-ozel-2015400123) [inline frontend/src/app/events/pages/event/event.page.ts]: Removed unused imports this and the one below

File: frontend/src/app/profile/pages/profile/profile.page.ts [hunk 5]
```diff
@@ -20,12 +21,16 @@ export class ProfilePage implements OnInit {
   private sub : any;
   user: User | null;
   userId : string | null;
-  constructor(private auth : AuthService, private router : Router,
-              private route: ActivatedRoute) { }
+  constructor(private auth : AuthService,
```
Comment (serkan-ozel-2015400123) [inline frontend/src/app/profile/pages/profile/profile.page.ts]: added user service and also organize long constructor to have multiple lines

File: frontend/src/app/profile/pages/public/public.page.ts [hunk 6]
```diff
@@ -19,32 +21,44 @@ export class PublicPage implements OnInit {
...
             this.error = true;
+            this.loading = false;
           }
         );
       }
     });
     if(this.auth.isAuthenticated()){
       this.signedInId = this.auth.getUserId();
-      if(this.signedInId == this.userId) this.sameUser = true;
+      if(this.signedInId == this.userId) {
```
Comment (serkan-ozel-2015400123) [inline frontend/src/app/profile/pages/public/public.page.ts]: increased readability for me. So added this change too

**Comment by srknzl** [comment]
anyone? @bounswe/bounswe2018group9

**Pull request closed without merging**